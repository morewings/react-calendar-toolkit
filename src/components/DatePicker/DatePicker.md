```js noeditor
import DatePicker from 'react-calendar-toolkit';

const style = {
  color: 'black',
  textAlign: 'center',
  padding: '8px 0',
  border: '2px dashed pink',
  margin: '2px'
};

const styleCalendar = {
  ...style,
  height: '225px'
};

const styleHeader = {
  ...style,
  height: '70px'
};

const styleSelector = {
  ...style,
  height: '26px'
};

const CustomComponent = props => {
  return (
    <div style={props.style}>
      {props.title}
    </div>
  )
};


<div style={{display: 'flex'}}>
  <DatePicker
    renderHeaderAs={() => <CustomComponent style={styleHeader} title="<Header />" />}
    renderSelectorAs={() => <CustomComponent style={styleSelector} title="<Selector />" />}
    wrapMonthWith={() => <CustomComponent style={styleCalendar} title="<Calendar />" />}
    wrapWeekDaysWith={() => <CustomComponent style={style} title="<Weekdays />" />}
    minPrecision="month"
    initialDate={new Date(2020, 1, 6)}
    startDate={new Date(2019, 11, 6)}
    endDate={new Date(2020, 1, 25)}
    today={new Date()}
    onDateSet={date => {console.log('date set', date);}}
    title="Demo datepicker" />
  <div style={{marginLeft: '30px'}}>
    <DatePicker
      minPrecision="month"
      initialDate={new Date(2020, 1, 6)}
      startDate={new Date(2019, 11, 6)}
      endDate={new Date(2020, 1, 25)}
      today={new Date()}
      onDateSet={
        date => {
          console.log('date set', date);
      }}
      title="Demo datepicker" />
  </div>
</div>
```

Datepicker is a component which allows user to see and pick from Calendar of provided dates. It has Selector, which allows to switch between precisions and step between months. Also renders Header with selected date and Datepicker title. 

```js
import DatePicker from 'react-calendar-toolkit';

<DatePicker
  initialDate={new Date(2020, 1, 6)}
  startDate={new Date(2019, 11, 6)}
  endDate={new Date(2020, 1, 25)}
  today={new Date()}
  onDateSet={date => {console.log('date set', date);}}
  title="Demo datepicker" />
```

### Basic Customization

#### No header
```js
import DatePicker from 'react-calendar-toolkit';

<DatePicker
  showHeader={false}
  onDateSet={date => {console.log('date set', date);}} />
```

#### Disable weekend highlight
```js
import DatePicker from 'react-calendar-toolkit';

<DatePicker
  highlightWeekends={false}
  onDateSet={date => {console.log('date set', date);}}/>
```

#### With custom locale

```js
import locale from 'date-fns/esm/locale/ru';
import DatePicker from 'react-calendar-toolkit';

<DatePicker
  dateFnsLocale={locale}
  onDateSet={date => {console.log('date set', date);}} />
```
You can provided any included ``date-fns`` locale. Or create a [custom one](https://date-fns.org/v2.9.0/docs/I18n-Contribution-Guide#adding-a-new-locale). First day of the week is taken from locale object, `locale.options.weekStartsOn`.

#### Disable and highlight dates
```js
import {isSameDay} from 'date-fns'
import DatePicker from 'react-calendar-toolkit';

const disableWeekends = ({isWeekend, precision}) => {
  // NB precision check, otherWise all months will be disabled in Month Calendar view
  return precision==='day' && isWeekend
};

const highlightDate = ({date, precision}) => {
  return isSameDay(date, new Date(2020, 0, 7))
};

<DatePicker
  disableDate={disableWeekends}
  highlightDate={highlightDate}
  onDateSet={date => {console.log('date set', date);}} />
```
You can disable (custom style and non-clickable) or highlight (custom style) any rendered date.

#### Define minimal precision of `Calendar`

Setting `minPrecision` prop allows to define what kind of date user is able to select. E. g. if `minPrecision === 'month''`, user can select only between _months_ and _years_. 

```js
import DatePicker from 'react-calendar-toolkit';

<DatePicker
  minPrecision="month"
  onDateSet={date => {console.log('date set', date);}} />
```

```js
import DatePicker from 'react-calendar-toolkit';

<DatePicker
      minPrecision="year"
      startDate={new Date(1990, 0, 1)}
      endDate={new Date(2020, 1, 25)}
      onDateSet={date => {console.log('date set', date);}} />
```

### UI customization
You can override wrapping element or className of `Datepicker` with `wrapWith` prop, which expects React.Element or DOM node. 

#### Wrap Datepicker with custom component

`Datepicker` `wrapWith` component wraps it with provided `DOM node` or `React.Element`. ALso in Default UI it's used to set CSS variables with `useThemePostCSS` hook, otherwise CSS variables will be undefined.

```js
import React, {useRef} from 'react';
import DatePicker, {useThemePostCSS} from 'react-calendar-toolkit';

const style = {
  fontFamily: 'sans-serif',
  width: '260px',
  border: '2px dashed pink',
};

const CustomComponent = props => {

  /** Hook to set style variables from Theme Context */
  const [ref] = useThemePostCSS();
  return (
    <div
      ref={ref}
      style={style}
      aria-label={props.title}
      role="dialog">
      {props.children}
    </div>
  );
};

<DatePicker
  onDateSet={date => {console.log('date set', date);}}
  minPrecision="day"
  wrapWith={CustomComponent}
  title="Demo datepicker" />
```

#### Default wrapper
```js { "file": "./../../../../src/components/visual/Datepicker/DatepickerWrapper.js" }
```

See **Provide custom UI** section for more info.