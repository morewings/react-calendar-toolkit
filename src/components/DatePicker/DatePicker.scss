@import "./../../variables";

.datepicker-wrapper{
  width: $datePickerWidth;
  border: 1px solid gray;
}