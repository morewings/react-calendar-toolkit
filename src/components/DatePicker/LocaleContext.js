import React from 'react';

const LocaleContext = React.createContext({});

export const useLocaleContext = () => {};

export default LocaleContext;
