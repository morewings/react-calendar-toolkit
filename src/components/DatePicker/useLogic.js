import {useCallback} from 'react';
import {
  useDatePickerActions,
  useDatePickerContext,
  useIncrementPrecision,
} from 'features/datepicker';

const useLogic = (onDateSet, minPrecision) => {
  const {
    state: {precision},
  } = useDatePickerContext();
  const {setDate, setVisibility} = useDatePickerActions();
  const incrementPrecision = useIncrementPrecision(minPrecision);
  const handleDateSet = useCallback(
    date => {
      if (precision === minPrecision) {
        onDateSet(date);
        setDate(date);
      } else {
        setVisibility(date);
        incrementPrecision();
      }
    },
    [
      incrementPrecision,
      minPrecision,
      onDateSet,
      precision,
      setDate,
      setVisibility,
    ]
  );
  return {handleDateSet};
};

export default useLogic;
