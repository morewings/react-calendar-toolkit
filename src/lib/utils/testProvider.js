import React from 'react';
import PropTypes from 'prop-types';
import {Provider as LocaleProvider} from 'lib/features/locale/withLocaleContext';
import {Provider as ThemeProvider} from 'lib/features/theme/withTheme';
import {Provider as ModalProvider} from 'lib/features/modal/withModalProvider';
import {Provider as DatepickerProvider} from 'lib/features/datepicker/withDatepickerProvider';

export const DatepickerMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <DatepickerProvider>{props.children}</DatepickerProvider>
    </ThemeProvider>
  </LocaleProvider>
);

export const InputMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <ModalProvider>
        <DatepickerProvider>{props.children}</DatepickerProvider>
      </ModalProvider>
    </ThemeProvider>
  </LocaleProvider>
);

DatepickerMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

InputMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
