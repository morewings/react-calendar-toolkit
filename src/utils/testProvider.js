import React from 'react';
import PropTypes from 'prop-types';
import {Provider as LocaleProvider} from 'features/locale/withLocaleContext';
import {Provider as ThemeProvider} from 'features/theme/withTheme';
import {Provider as ModalProvider} from 'features/modal/withModalProvider';
import {Provider as DatepickerProvider} from 'features/datepicker/withDatepickerProvider';

export const DatepickerMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <DatepickerProvider>{props.children}</DatepickerProvider>
    </ThemeProvider>
  </LocaleProvider>
);

export const InputMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <ModalProvider>
        <DatepickerProvider>{props.children}</DatepickerProvider>
      </ModalProvider>
    </ThemeProvider>
  </LocaleProvider>
);

DatepickerMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

InputMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
