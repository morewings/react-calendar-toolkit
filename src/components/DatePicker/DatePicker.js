import React, {useCallback, useEffect, Fragment} from 'react';
import PropTypes from 'prop-types';
import {useDispatch, useSelector} from 'react-redux';
import {convertToTimestamp, convertToDate} from 'utils/dateUtils';
import config from 'utils/config';
import {selectors, actionCreators} from 'features/datepicker';
import Calendar from 'components/logic/Calendar';
import WeekDays from 'components/logic/Weekdays';
import Selector from 'components/logic/Selector';
import DatepickerWrapper from 'components/visual/Datepicker';
import Header from 'components/visual/Header';
import SelectorVisual from 'components/visual/Selector';
import Day, {DayGrid} from 'components/visual/Day';
import Month, {MonthGrid} from 'components/visual/Month';
import Year, {YearGrid} from 'components/visual/Year';
import WeekDay, {WeekDayGrid} from 'components/visual/WeekDay';

const getNextPrecision = (precisionEnum, currentPrecision) => {
  const currentIndex = precisionEnum.indexOf(currentPrecision);
  return precisionEnum[currentIndex + 1];
};

const limitPrecision = (precisionEnum, minPrecision) => {
  const currentIndex = precisionEnum.indexOf(minPrecision);
  return precisionEnum.slice(0, currentIndex + 1);
};

const DatePicker = ({
  initialDate,
  today,
  showHeader,
  title,
  minPrecision,
  onDateSet,
  renderDayAs,
  wrapDaysWith,
  renderMonthAs,
  wrapMonthWith,
  renderYearAs,
  wrapYearWith,
  renderWeekDayAs,
  wrapWeekDaysWith,
  startDate,
  endDate,
  disableDate,
  highlightWeekends,
  renderHeaderAs,
  renderSelectorAs,
  highlightDate,
  wrapWith,
}) => {
  const dispatch = useDispatch();
  const selectedTimestamp = useSelector(selectors.getSelectedTimestamp);
  const todayTimestamp = useSelector(selectors.getTodayTimestamp);
  const visibleTimestamp = useSelector(selectors.getVisibleTimestamp);

  const precision = useSelector(selectors.getPrecision);
  const datepickerPrecisions = limitPrecision(
    config.supportedPrecisions,
    minPrecision
  );
  const nextPrecision = getNextPrecision(datepickerPrecisions, precision);

  // TODO: refactor to hook
  const handleDateSet = useCallback(
    date => {
      precision === minPrecision && onDateSet(date);
      dispatch(actionCreators.setDate(date));
      nextPrecision && dispatch(actionCreators.setPrecision(nextPrecision));
    },
    [dispatch, minPrecision, nextPrecision, onDateSet, precision]
  );

  useEffect(() => {
    dispatch(actionCreators.setDate(initialDate));
    dispatch(actionCreators.setVisibility(initialDate));
  }, [dispatch, initialDate]);

  useEffect(() => {
    dispatch(actionCreators.setToday(today));
    dispatch(actionCreators.setPrecision(minPrecision));
  }, [dispatch, minPrecision, today]);

  const Wrapper = wrapWith;

  const HeaderUI = renderHeaderAs;
  const SelectorUI = renderSelectorAs;

  return (
    <Wrapper title={title}>
      {showHeader && (
        <HeaderUI
          title={title}
          todayDate={convertToDate(todayTimestamp)}
          selectedDate={convertToDate(selectedTimestamp)}
        />
      )}
      <Selector
        renderAs={props => (
          <SelectorUI minPrecision={minPrecision} {...props} />
        )}
        selectedTimestamp={selectedTimestamp}
        todayTimestamp={todayTimestamp}
        visibleTimestamp={visibleTimestamp}
        startDate={startDate}
        endDate={endDate}
      />
      {precision === 'day' && (
        <Fragment>
          <WeekDays renderAs={renderWeekDayAs} wrapWith={wrapWeekDaysWith} />
          <Calendar
            precision="day"
            highlightWeekends={highlightWeekends}
            wrapWith={wrapDaysWith}
            renderAs={renderDayAs}
            disableDate={disableDate}
            highlightDate={highlightDate}
            selectedTimestamp={selectedTimestamp}
            visibleTimestamp={visibleTimestamp}
            todayTimestamp={todayTimestamp}
            onDateSet={handleDateSet}
            startDate={startDate}
            endDate={endDate}
          />
        </Fragment>
      )}
      {precision === 'month' && (
        <Calendar
          precision="month"
          disableDate={disableDate}
          highlightDate={highlightDate}
          wrapWith={wrapMonthWith}
          renderAs={renderMonthAs}
          selectedTimestamp={selectedTimestamp}
          todayTimestamp={todayTimestamp}
          onDateSet={handleDateSet}
          startDate={startDate}
          endDate={endDate}
        />
      )}
      {precision === 'year' && (
        <Calendar
          precision="year"
          disableDate={disableDate}
          highlightDate={highlightDate}
          wrapWith={wrapYearWith}
          renderAs={renderYearAs}
          selectedTimestamp={selectedTimestamp}
          todayTimestamp={todayTimestamp}
          onDateSet={handleDateSet}
          startDate={startDate}
          endDate={endDate}
        />
      )}
    </Wrapper>
  );
};

DatePicker.propTypes = {
  /** Set initial selected date when component renders. */
  initialDate: PropTypes.instanceOf(Date),
  /** Set today date. */
  today: PropTypes.instanceOf(Date),
  /** Set start date of calendar. */
  startDate: PropTypes.instanceOf(Date),
  /** Set end date of calendar. */
  endDate: PropTypes.instanceOf(Date),
  /** Flag to show or hide header. */
  showHeader: PropTypes.bool,
  /** Set title of calendar show in Header. */
  title: PropTypes.string,
  /** Set minimum precision (measuring unit) of calendar. Possible values: 'day', 'month', 'year'. */
  minPrecision: PropTypes.oneOf(['year', 'month', 'day']),
  /** Callback when user clicks selected date */
  onDateSet: PropTypes.func.isRequired,
  /** Define component which wraps __Datepicker__. */
  wrapWith: PropTypes.elementType,
  /** Define component which renders __day__ entry. */
  renderDayAs: PropTypes.elementType,
  /** Define component which wraps __day__ entry. */
  wrapDaysWith: PropTypes.elementType,
  /** Define component which renders __month__ entry. */
  renderMonthAs: PropTypes.elementType,
  /** Define component which wraps __month__ entry. */
  wrapMonthWith: PropTypes.elementType,
  /** Define component which renders __year__ entry. */
  renderYearAs: PropTypes.elementType,
  /** Define component which wraps __year__ entry. */
  wrapYearWith: PropTypes.elementType,
  /** Define component which renders __week day__ entry. */
  renderWeekDayAs: PropTypes.elementType,
  /** Define component which wraps __week day__ entry. */
  wrapWeekDaysWith: PropTypes.elementType,
  /** Define component which renders __Header__. */
  renderHeaderAs: PropTypes.elementType,
  /** Define component which renders __Precision selector__. */
  renderSelectorAs: PropTypes.elementType,
  /** Function which decides if date should be __disabled__. It gets `isWeekend`, `precision` and `date` and outputs Boolean.
   * `({isWeekend, precision, date}) => false`
   * */
  disableDate: PropTypes.func,
  /** Function which decides if date should be __highlighted__. It gets `isWeekend`, `precision` and `date` and outputs Boolean.
   * `({isWeekend, precision, date}) => false`
   * */
  highlightDate: PropTypes.func,
  /** Flag to enable __weekend highlight__ prop. */
  highlightWeekends: PropTypes.bool,
  /** date-fns locale object. Defaults to english */
  dateFnsLocale: PropTypes.shape({}), // eslint-disable-line react/require-default-props
  /** Theme object to customize style for UI components */
  theme: PropTypes.shape({}), // eslint-disable-line react/require-default-props
};

DatePicker.defaultProps = {
  initialDate: new Date(2020, 0, 6),
  startDate: new Date(2020, 0, 1),
  endDate: new Date(2020, 1, 25),
  today: new Date(),
  showHeader: true,
  title: '',
  minPrecision: 'day',
  wrapWith: DatepickerWrapper,
  renderDayAs: Day,
  wrapDaysWith: DayGrid,
  renderMonthAs: Month,
  wrapMonthWith: MonthGrid,
  renderYearAs: Year,
  wrapYearWith: YearGrid,
  renderWeekDayAs: WeekDay,
  wrapWeekDaysWith: WeekDayGrid,
  renderHeaderAs: Header,
  renderSelectorAs: SelectorVisual,
  disableDate: ({isWeekend, precision, date}) => false,
  highlightDate: ({isWeekend, precision, date}) => false,
  highlightWeekends: true,
};

export default DatePicker;
