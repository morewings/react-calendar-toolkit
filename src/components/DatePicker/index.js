import {compose} from 'redux';
import DatePicker from './DatePicker';
import {withLocaleContext} from '../../utils/localeContext';
import {withTheme} from '../../utils/themeContext';

export default compose(withLocaleContext, withTheme)(DatePicker);
