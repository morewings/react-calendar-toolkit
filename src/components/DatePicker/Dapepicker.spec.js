import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {DatepickerMockProvider as Provider} from 'utils/testProvider';
import DatePicker from './DatePicker';

const onDateSet = jest.fn();

describe('DatePicker', () => {
  beforeEach(() => {
    onDateSet.mockClear();
  });

  it('renders', () => {
    const {asFragment} = render(<DatePicker onDateSet={onDateSet} />, {
      wrapper: ({children}) => <Provider>{children}</Provider>,
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
