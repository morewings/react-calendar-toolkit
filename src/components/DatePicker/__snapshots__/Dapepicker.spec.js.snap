// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 30px; --entryMargin: 2px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div
      class="wrapper"
    >
      <div
        class="title"
      />
      <div
        aria-live="polite"
        class="currentDate"
      >
        Jan 6th
      </div>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="buttons"
      >
        <button
          aria-live="polite"
          type="button"
        >
          2020
        </button>
        <button
          aria-live="polite"
          type="button"
        >
          January
        </button>
      </div>
      <div
        class="stepper"
      >
        <button
          aria-label="1 month ago"
          type="button"
        >
          ⟨
        </button>
        <button
          aria-label="in 1 month"
          type="button"
        >
          ⟩
        </button>
      </div>
    </div>
    <div
      class="grid"
      role="presentation"
    >
      <div
        aria-label="Sunday"
        class="wrapper"
      >
        Su
      </div>
      <div
        aria-label="Monday"
        class="wrapper"
      >
        Mo
      </div>
      <div
        aria-label="Tuesday"
        class="wrapper"
      >
        Tu
      </div>
      <div
        aria-label="Wednesday"
        class="wrapper"
      >
        We
      </div>
      <div
        aria-label="Thursday"
        class="wrapper"
      >
        Th
      </div>
      <div
        aria-label="Friday"
        class="wrapper"
      >
        Fr
      </div>
      <div
        aria-label="Saturday"
        class="wrapper"
      >
        Sa
      </div>
    </div>
    <div
      class="grid"
    >
      <div
        class="placeholder"
      />
      <div
        class="placeholder"
      />
      <div
        class="placeholder"
      />
      <div
        aria-disabled="false"
        aria-label="1st January Wednesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        1
      </div>
      <div
        aria-disabled="false"
        aria-label="2nd January Thursday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        2
      </div>
      <div
        aria-disabled="false"
        aria-label="3rd January Friday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        3
      </div>
      <div
        aria-disabled="false"
        aria-label="4th January Saturday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        4
      </div>
      <div
        aria-disabled="false"
        aria-label="5th January Sunday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        5
      </div>
      <div
        aria-disabled="false"
        aria-label="6th January Monday"
        class="wrapper isSelected"
        role="button"
        tabindex="0"
      >
        6
      </div>
      <div
        aria-disabled="false"
        aria-label="7th January Tuesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        7
      </div>
      <div
        aria-disabled="false"
        aria-label="8th January Wednesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        8
      </div>
      <div
        aria-disabled="false"
        aria-label="9th January Thursday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        9
      </div>
      <div
        aria-disabled="false"
        aria-label="10th January Friday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        10
      </div>
      <div
        aria-disabled="false"
        aria-label="11th January Saturday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        11
      </div>
      <div
        aria-disabled="false"
        aria-label="12th January Sunday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        12
      </div>
      <div
        aria-disabled="false"
        aria-label="13th January Monday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        13
      </div>
      <div
        aria-disabled="false"
        aria-label="14th January Tuesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        14
      </div>
      <div
        aria-disabled="false"
        aria-label="15th January Wednesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        15
      </div>
      <div
        aria-disabled="false"
        aria-label="16th January Thursday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        16
      </div>
      <div
        aria-disabled="false"
        aria-label="17th January Friday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        17
      </div>
      <div
        aria-disabled="false"
        aria-label="18th January Saturday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        18
      </div>
      <div
        aria-disabled="false"
        aria-label="19th January Sunday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        19
      </div>
      <div
        aria-disabled="false"
        aria-label="20th January Monday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        20
      </div>
      <div
        aria-disabled="false"
        aria-label="21st January Tuesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        21
      </div>
      <div
        aria-disabled="false"
        aria-label="22nd January Wednesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        22
      </div>
      <div
        aria-disabled="false"
        aria-label="23rd January Thursday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        23
      </div>
      <div
        aria-disabled="false"
        aria-label="24th January Friday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        24
      </div>
      <div
        aria-disabled="false"
        aria-label="25th January Saturday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        25
      </div>
      <div
        aria-disabled="false"
        aria-label="26th January Sunday"
        class="wrapper isWeekend"
        role="button"
        tabindex="0"
      >
        26
      </div>
      <div
        aria-disabled="false"
        aria-label="27th January Monday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        27
      </div>
      <div
        aria-disabled="false"
        aria-label="28th January Tuesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        28
      </div>
      <div
        aria-disabled="false"
        aria-label="29th January Wednesday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        29
      </div>
      <div
        aria-disabled="false"
        aria-label="30th January Thursday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        30
      </div>
      <div
        aria-disabled="false"
        aria-label="31st January Friday"
        class="wrapper"
        role="button"
        tabindex="0"
      >
        31
      </div>
      <div
        class="placeholder"
      />
    </div>
  </div>
</DocumentFragment>
`;
